### ISSUE TITLE
unify valueType and phetioType options

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/axon#194]
While we're doing https://github.com/phetsims/axon/issues/189 (Simplify Property type validation) ...

Let's consider unifying `valueType` and `phetioType` options.  They are currently 2 very different ways of expressing identical semantics. Duplicating this information is undesirable for both instrumentation and maintainability.

Examples...

GRAPHING_QUADRATICS/GQModel

```js
      // @public {Property.<Quadratic|null>} the saved quadratic, null if nothing is saved
      this.savedQuadraticProperty = new Property( null, {
        isValidValue: value => ( value instanceof Quadratic || value === null ),
        ...
        phetioType: PropertyIO( NullableIO( QuadraticIO ) ),
        ...
      } );
```

GRAPHING_QUADRATICS/PointTool

```js
      // @public {Vector2}
      this.locationProperty = new Property( options.location, {
        valueType: Vector2,
        ...
        phetioType: PropertyIO( Vector2IO ),
        ...
      } );
```

AXON/BooleanProperty (and similarly in NumberProperty and StringProperty)

```js
  function BooleanProperty( value, options ) {

    options = _.extend( {
      phetioType: PropertyIO( BooleanIO )
    }, options );

    ...
    options.valueType = 'boolean';  // BooleanProperty requires values to be primitive booleans

    Property.call( this, value, options );
  }
```

**Comment by zepumph** [comment]
Wow what a great issue to create! We are looking into this for Emitter right now and it is going very well, see https://github.com/phetsims/axon/issues/190. To decide the main schema for determining what a "valid value" is, we will need to wait on the decision in https://github.com/phetsims/axon/issues/189

**Comment by pixelzoom** [comment]
See also https://github.com/phetsims/axon/issues/195, proper handling of `valueType: Array`.

**Comment by samreid** [comment]
On hold until #189 is complete.

**Comment by samreid** [comment]
See also #204, which is the same issue for Emitters.

**Comment by pixelzoom** [comment]
It seems like we addressed this in https://github.com/phetsims/axon/issues/189, where we concluded that these options will _not_ be unified.  @samreid @zepumph Can this issue be closed?

**Comment by samreid** [comment]
That sounds right to me, leaving up to @zepumph in case he has other thoughts or recommendations.

**Comment by zepumph** [comment]
>where we concluded that these options will not be unified.

I think that you and I understand this in the same way @pixelzoom, though I want state the decision for this issue. Though they won't be unified via a mutual exclusion assertion, phetioType does cover `valueType` in most cases and, by convention, can/should be removed when the `phetioType` is provided. For example `phetioType: NullableIO( BooleanIO )` is a superset of information of `valueType: [ null, 'boolean' ]`, and so the `valueType` doesn't need to also be provided (and can be removed if already supplied). Closing
